### PR TITLE
deploy: Use glnx file copy code

### PR DESCRIPTION
**Pull request by cgwalters**
This is centralizing more generic utility code in libglnx.

### CODE REVIEW

**Comment by cgwalters** [comment]
Depends on https://github.com/GNOME/libglnx/pull/3

**Comment by cgwalters** [comment]
Added some more cleanup commits here.

File: src/libostree/ostree-repo-commit.c [hunk 1]
```diff
@@ -848,65 +848,57 @@ devino_equal (gconstpointer   a,
...
-                                              error);
-      if (!enumerator)
+      if (!glnx_dirfd_iterator_next_dent (&dfd_iter, &dent, cancellable, error))
         goto out;
+          
+      if (dent == NULL)
+        break;
 
-      dirname = gs_file_get_basename_cached (objdir);
+      if (strlen (dent->d_name) != 2)
```
Comment (mbarnes) [inline src/libostree/ostree-repo-commit.c]: This `strlen` check could use a comment.  I think it's to recurse only in the two-digit hex directories, right?

File: src/libostree/ostree-repo-commit.c [hunk 2]
```diff
@@ -921,17 +913,26 @@ scan_loose_devino (OstreeRepo                     *self,
 
           dot = strrchr (name, '.');
           g_assert (dot);
-
+          
           if ((dot - name) != 62)
```
Comment (mbarnes) [inline src/libostree/ostree-repo-commit.c]: Suggest a comment here as well.  This is the rest of the 64-digit hash, I assume.

**Comment by mbarnes** [comment]
Couple spots to clarify (if only for my sake), but overall looks like a nice cleanup.  :+1:

**Comment by cgwalters** [comment]
Thanks, pushed with added comments there.

**Pull request closed without merging**